ArcGISRuntimeSDKQt_CppSamples/Layers/DisplaySubtypeFeatureLayer/DisplaySubtypeFeatureLayer.cpp

```diff
@@ -0,0 +1,147 @@
+  QObject(parent),
+  m_map(new Map(Basemap::streetsNightVector(this), this)),
+  m_alternateRenderer(new SimpleRenderer(new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Diamond, QColor(Qt::magenta), 20, this), this)),
+  m_labelJson(QStringLiteral("{ \"labelExpression\":\"[nominalvoltage]\",\"labelPlacement\":\"esriServerPointLabelPlacementAboveRight\",\"useCodedValues\":true,\"symbol\":{\"angle\":0,\"backgroundColor\":[0,0,0,0],\"borderLineColor\":[0,0,0,0],\"borderLineSize\":0,\"color\":[0,0,255,255],\"font\":{\"decoration\":\"none\",\"size\":10.5,\"style\":\"normal\",\"weight\":\"normal\"},\"haloColor\":[255,255,255,255],\"haloSize\":2,\"horizontalAlignment\":\"center\",\"kerning\":false,\"type\":\"esriTS\",\"verticalAlignment\":\"middle\",\"xoffset\":0,\"yoffset\":0}}"))
+{
+  m_busy = true;
+  emit busyChanged();
+  // create the feature table
+  ServiceFeatureTable* featureTable = new ServiceFeatureTable(QUrl("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100"), this);
+  // create the feature layer using the feature table
+  m_subtypeFeatureLayer = new SubtypeFeatureLayer(featureTable, this);
+
+  // add the feature layer to the map
+  m_map->operationalLayers()->append(m_subtypeFeatureLayer);
+
+  // set the viewpoint to Naperville, Illinois
+  m_map->setInitialViewpoint(Viewpoint(Envelope(-9812691.11079696, 5128687.20710657, -9812377.9447607, 5128865.36767282, SpatialReference::webMercator())));
+
+  // when subtype feature layer is loaded get the subtype sublayer street lights and define it's labels
+  connect(m_subtypeFeatureLayer, &SubtypeFeatureLayer::doneLoading, this, &DisplaySubtypeFeatureLayer::getSubtypeSublayerAndDefineLabels);
+
+  connect(m_map, &Map::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    connect(m_mapView, &MapQuickView::mapScaleChanged, this, [this]()
+    {
+      m_mapScale = m_mapView->mapScale();
+      emit mapScaleChanged();
+    });
+  });
+}
+
+DisplaySubtypeFeatureLayer::~DisplaySubtypeFeatureLayer() = default;
+
+void DisplaySubtypeFeatureLayer::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplaySubtypeFeatureLayer>("Esri.Samples", 1, 0, "DisplaySubtypeFeatureLayerSample");
+}
+
+MapQuickView* DisplaySubtypeFeatureLayer::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DisplaySubtypeFeatureLayer::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void DisplaySubtypeFeatureLayer::getSubtypeSublayerAndDefineLabels(const Error& e)
+{
+  if (!e.isEmpty())
+    return;
+
+  m_busy = false;
+  emit busyChanged();
+
+  // get the Street Light sublayer and define its labels
+  m_subtypeSublayer = m_subtypeFeatureLayer->sublayerWithSubtypeName("Street Light", this);
```

Comment:
is there any scenario `m_subtypeFeatureLayer` could be null here?

Reply:
Yes. if the name passed in as a parameter is not found it will return null i believe. That's why I have added checks for the renderer as well as setting the min scale.

Reply:
@jared-esri my concern is calling `m_subtypeFeatureLayer->sublayerWithSubtypeName()` without checking if `m_subtypeFeatureLayer` is nullptr or not first. Wasn't sure if there is any guarantee that this would be not null by the time this was called. If not, I suggest checking if it's null before accessing the sublayerWithSubtypeName

Reply:
Oh I see now. Yes there is no guarantee it will not be null. I will make the change.